linode/manager
# refactor: [M3-6256] - Provide a Typescript type for `api-v4` params and filters

## Description 📝

- Adds a typescript type for api-v4's params and filters
- Removes Redux for Firewalls because firewalls are in React Query
- Removes `getAllWithArguments` because it was confusing

## How to test 🧪

- Verify typecheck passes
- Look at the type definition for `Params` and `Filter` and let me know if you have any improvements that could be made
